core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -372,6 +372,8 @@ object SparkSubmit {
       OptionAssigner(args.files, YARN, CLUSTER, clOption = "--files"),
       OptionAssigner(args.archives, YARN, CLUSTER, clOption = "--archives"),
       OptionAssigner(args.jars, YARN, CLUSTER, clOption = "--addJars"),
+      OptionAssigner(args.principal, YARN, CLUSTER, clOption = "--principal"),
```

Comment:
I understand these options are mostly useful in cluster mode, but is there any technical reason why they shouldn't be exposed in client mode too?